instrumentation/netty-reactor-0.8.0/src/main/java/reactor/core/scheduler/Schedulers_Instrumentation.java

```diff
@@ -0,0 +1,45 @@
+/*
+ *  Copyright 2021 New Relic Corporation. All rights reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+package reactor.core.scheduler;
+
+import com.newrelic.api.agent.weaver.MatchType;
+import com.newrelic.api.agent.weaver.Weave;
+import com.newrelic.api.agent.weaver.Weaver;
+import com.nr.instrumentation.reactor.netty.TokenLinkingSubscriber;
+import reactor.core.publisher.Hooks;
+import reactor.core.publisher.Hooks_Instrumentation;
+
+import static com.nr.instrumentation.reactor.netty.TokenLinkingSubscriber.tokenLift;
+
+@Weave(type = MatchType.BaseClass, originalName = "reactor.core.scheduler.Schedulers")
+public abstract class Schedulers_Instrumentation {
+
+    @Weave(type = MatchType.ExactClass, originalName = "reactor.core.scheduler.Schedulers$CachedScheduler")
+    static class CachedScheduler {
+        final Scheduler cached;
+        final String key;
+
+        CachedScheduler(String key, Scheduler cached) {
+            /*
+             * Add tokenLift hook if it hasn't already been added. This allows for tokens to be retrieved from
+             * the current context and linked across threads at various points of the Flux/Mono lifecycle.
+             *
+             * When using Netty Reactor with SpringBoot this hook will be added by the HttpTrafficHandler_Instrumentation
+             * but when using other embedded web servers (e.g. Tomcat, Jetty, Undertow) the HttpTrafficHandler class
```

Comment:
I feel like there is another piece of relevant piece of work here (for the future, after this PR). 

If the question comes up and people wonder if this instrumentation works for a non-netty server, I've seen it does from your tests.  It could be good to create an issue to confirm this behavior for a 1-2 most popular non-netty servers and create AITS for them.

Reply:
Agreed! I'm hoping to look into that before wrapping up this work.

Reply:
@XiXiaPdx This should do it.

This [PR](https://source.datanerd.us/java-agent/agent-integration-tests/pull/65) updates the AITs to test Spring WebFlux with two different SpringBoot apps.
- SpringBoot configured with Spring Security and Reactor Netty as the embedded web server
- SpringBoot configured with Spring Security and Tomcat as the embedded web server